src/slurm_plugin/clustermgtd.py

```diff
@@ -895,17 +701,18 @@ def _increase_partitions_protected_failure_count(self, bootstrap_failure_nodes):
                 )
 
     @log_exception(log, "maintaining unhealthy dynamic nodes", raise_on_error=False)
-    def _handle_unhealthy_dynamic_nodes(self, unhealthy_dynamic_nodes, private_ip_to_instance_map):
+    def _handle_unhealthy_dynamic_nodes(self, unhealthy_dynamic_nodes):
         """
         Maintain any unhealthy dynamic node.
 
         Terminate instances backing dynamic nodes.
         Setting node to down will let slurm requeue jobs allocated to node.
         Setting node to power_down will terminate backing instance and reset dynamic node for future use.
         """
-        instances_to_terminate = ClusterManager._get_backing_instance_ids(
-            unhealthy_dynamic_nodes, private_ip_to_instance_map
-        )
+        instances_to_terminate = []
+        for node in unhealthy_dynamic_nodes:
+            if node.instance and node.instance.id:
+                instances_to_terminate.append(node.instance.id)
```

Comment:
1-liner: `instances_to_terminate += [ node.instance.id for node in unhealthy_dynamic_nodes and node.instance ]`